### DISCUSSION TITLE
get TypeError: expected str, got BatchTracer when using jax.debug.print

### DISCUSSION THREAD

**Discussion by julia-lu-2019** [jax-ml/jax#20559] [Q&A] [answered]
I want to print the numerical value of the first batch of the input using jax.debug.print，however I get TypeError: expected str, got BatchTracer. Does anyone know how to achieve this without triggering error?
here is my code:
`def slater_determinants(params, x):

    # x.shape = (n_elec * input_dim,)
    x = x.reshape(n_elec, input_dim)
    for param in params:
        # Each electron is processed by the same MLP function
        x = jnp.matmul(x, param)
        jx.debug.print(x.value[0])
        x = jnp.tanh(x)

    x = x + jnp.eye(x.shape[0])

    _, lnpsi = jnp.linalg.slogdet(x)
    return lnpsi`

the following is the report of the error:
`  File "/root/anaconda3/envs/jax/lib/python3.9/string.py", line 259, in parse
    return _string.formatter_parser(format_string)
TypeError: expected str, got BatchTracer`

**Comment by jakevdp** [comment, marked as answer]
The first argument of `jax.debug.print` must be a format string, so for example you could do this:
```python
jx.debug.print("{}", x.value[0])
```
See the [debug.print docs](https://jax.readthedocs.io/en/latest/debugging/print_breakpoint.html#debugging-with-jax-debug-print-and-other-debugging-callbacks) for more examples.

**Reply by julia-lu-2019** [reply]
thx! The answer really helps solve my problem.
